docs/proposals/parquet-storage.md

```diff
@@ -0,0 +1,139 @@
+- Index relies on postings, which can be a huge bottleneck for high cardinality data
+
+### Store Gateway Operational Challenges
+
+Store Gateway is originally introduced in [Thanos](https://thanos.io/). Both Cortex and Thanos community have been collaborating to add a lot of optimizations to Store Gateway. However, it has its own problems related to the design.
+
+1. Resource Intensive
+   - Requires significant local disk space to store index headers
+   - High memory utilization due to index header mmap
+   - Often needs over-provisioning to handle query spikes
+
+2. State Management and Scaling Difficulties
+   - Requires complex data sharding when scaling. Often causing issues such as consistency check failure. Hard to configure for users
+   - Initial sync causes long startup time. This affects service availability on both scaling and failure recovery scenario
+
+3. Query Inefficiencies
+   - Attempts to minimize storage requests often lead to overfetching, causing high bandwidth usage
+   - Complex caching logic with varying effectiveness. Latency varies a lot when cache miss
+   - Processes single block with one goroutine, leading to high latency for large blocks and cannot scale without complex data partitioning
+
+### Why Parquet?
+
+[Apache Parquet](https://parquet.apache.org/) is a columnar storage format designed specifically for efficient data storage and retrieval from object storage systems. It offers several key advantages that directly address the problems we face with TSDB and Store Gateway:
+
+- Data organized by columns rather than rows, reduces number of requests to object storage as only limited IO is required to fetch the whole column
+- Rich file metadata and index, no local state like index header required to query the data, making it stateless
+- Advanced compression techniques reduce storage costs and improve query performance
+- Parallel processing friendly using Parquet Row Group
+
+There are other benefits of Parquet formats, but they are not directly related to the proposal:
+
+- Wide ecosystem and tooling support
+- Column pruning opportunity using projection pushdown
+
+## Out of Scope
+
+- Allow Ingester and Compactor to create Parquet files instead of TSDB blocks directly. This could be in the future roadmap but this proposal only focuses on converting and querying Parquet files.
+
+## Proposed Design
+
+### Components
+
+There are 2 new Cortex components/modules introduced in this design.
+
+#### 1. Parquet Converter
+
+Parquet Converter is a new component that converts TSDB blocks on object store to Parquet file format.
+
+It is similar to compactor, however, it only converts single block. The converted Parquet files will be stored in the same TSDB block folder so that the lifecycle of Parquet file will be managed together with the block.
+
+Only certain blocks can be configured to convert to Parquet file and it can be block duration based, for example we only convert if block duration is >= 12h.
```

Comment:
Maybe we could add a sentence on why we wouldn't want to automatically convert all blocks to the Parquet format? :thinking: